### PR TITLE
Test for Tictactoe Game

### PR DESCRIPTION
**Pull request by Zubenna**
## In this pull request, we implemented the following;
- Wrote test for vertical win
- Wrote test for horizontal win
- Wrote test for diagonal win
- Wrote test to determine a tie game
- Wrote test to determine the valid choice
- Wrote test to determine taken choice

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+describe Board do
+  let(:gameboard) { Board.new }
+  describe '#play_again' do
+    it 'should output an array of 9 dashes' do
+      expect(gameboard.play_again).to eq(%w[- - - - - - - - -])
+    end
+  end
+
+  describe '#horizontal_winner?(player)' do
+    it 'should return true when get a horizontal line of symbol X' do
```
Comment (Forison) [inline spec/board_spec.rb]: * [ ] Currently, the test for horizontal winner captures only one situation thus ```1,2,3``` very well, but is silent on position ```4,5,6``` and ```7,8,9```. Kindly add the situation whereby the winner is at position ```4,5,6``` or ```7,8,9``` to match the implementation of the method.

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    it 'should return false when not get a horizontal line of symbol X' do
+      gameboard.update(1, 'Y')
+      gameboard.update(2, 'X')
+      gameboard.update(3, 'Y')
+      expect(gameboard.horizontal_winner?('X')).to eq(false)
+    end
+  end
+
+  describe '#vertical_winner?(player)' do
+    it 'should return true when get a vertical line of symbol X' do
```
Comment (Forison) [inline spec/board_spec.rb]: * [ ] Currently, the test for vertical winner captures only one situation thus ```1,4,7``` very well, but is silent on position ```2,5,8``` and ```3,6,9```. Kindly add the situation whereby the winner is at position ```2,5,8```, and ```3,6,9``` to match the implementation of the method.

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    it 'should return false when not get a vertical line of symbol X' do
+      gameboard.update(1, 'Y')
+      gameboard.update(4, 'X')
+      gameboard.update(7, 'Y')
+      expect(gameboard.vertical_winner?('X')).to eq(false)
+    end
+  end
+
+  describe '#diagonal_winner?(player)' do
+    it 'should return true when get a diagonal line of symbol X' do
```
Comment (Forison) [inline spec/board_spec.rb]: * [ ] Currently, the test for vertical winner captures only one situation thus ```1,5,9``` very well, but is silent on position ```3,5,7```. Kindly add the situation whereby the winner is at position ```3,5,7``` to match the implementation of the method.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes Required❌ 
##🙋‍♂️ Team!!!
#### You have done an awesome job with your test, however, here are few additions and recertification we have to make to get the project approved.
* [ ] Kindly add how to run the test in the readme file, this is very important especially when the focus of this task is testing.
* [ ] Check the comments 👇below👇 for the changes needed to have the project approved.
> The reason to have no print or put outside the main.rb was to have a robust test, the display method should have been in the main.rb, as required. It was one the requirement in the build-up milestone, I think that was milestone 3.
## That is it, kindly make the changes and submit the project for re-review👨‍🔧 .
#### Happy coding
Addo forison

**Review by mosesogwo — APPROVED** [review]
**Hi @Zubenna, @maosan132. Your project is approved. :heavy_check_mark:**

**Pull request merged**